tools/migration/cmd/go-filecoin-migrate.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/filecoin-project/go-filecoin/tools/migration/internal"
+	"github.com/filecoin-project/go-filecoin/tools/migration/migrate_1-to-2"
+)
+
+type Migrator interface {
+	Migrate(oldRepo, newRepo *os.File) error
+	Describe()
+	Validate(oldRepo, newRepo *os.File) error
+}
+
+// Run runs the given migration command
+func Run(cmd string, verbose bool) error {
+	var err error
+	miglog, err := makeMigl()
+	logpath := "/tmp/migration_log.txt"
+
+	oldRepo, err := getOldRepo()
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+
+	// open repodir read-only
+	// create new repodir & open read-write
+	// TODO: decide where to put the temp new repo.
+	newRepo, err := os.Create("/tmp/newRepo")
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+
+	var migrator Migrator
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+	// TODO see Issue #2585
+	migrator = migrate_1_to_2.NewMigrator_1_2(miglog)
+
+	switch cmd {
+	case "describe":
+		migrator.Describe()
+	case "buildonly":
+		err = migrator.Migrate(oldRepo, newRepo)
+	case "install":
+		if err = install(oldRepo, newRepo); err != nil {
+			miglog.Error(err)
+		}
+	case "migrate":
+		if err = migrator.Migrate(oldRepo, newRepo); err != nil {
+			miglog.Error(err)
+		}
+
+		if err = migrator.Validate(oldRepo, newRepo); err != nil {
+			miglog.Error(err)
+		}
+
+		if err = install(oldRepo, newRepo); err != nil {
+			miglog.Error(err)
+		}
+	}
+	if err != nil {
+		return err
+	}
+	miglog.Print(fmt.Sprintf("Done.  Please see log for more detail: ", logpath))
+	return nil
+}
+
+// swap old repo for new
+func install(oldRepo, newRepo *os.File) error {
+	return nil
+}
+
+// make a new logger
+func makeMigl() (*internal.Migl, error) {
+	logfile, err := os.Create("/tmp/foo.txt") // truncates
+	if err != nil {
+		return nil, err
+	}
+	migl := internal.NewMigl(logfile, true)
+	return &migl, nil
+}
+
+func getOldRepo() (*os.File, error) {
+	var oldRepo *os.File
+	var oldpath string
+	filpath := os.Getenv("FIL_PATH")
+	if filpath == "" {
+		homeDir := os.Getenv("HOME")
+		oldpath = fmt.Sprintf("%s/.filecoin", homeDir)
+	} else {
+		oldpath = filpath
+	}
+	oldRepo, err := os.Open(oldpath)
```

Comment:
Blocking: this must be configurable. Inject this environment lookup default from up in `main`.

Reply:
I ask that this part be out of scope. This PR was meant to just be a foundation for implementing features more fully, and if we wait until everything is fully implemented then itw will take longer to split the work with others.  E.g. IMO the temp migration directory ought to be configurable too eventually, but for the first go-round we are aiming for simplicity.

Reply:
Never mind -- it's in #2614 , more or less.